projects/pgai/pgai/vectorizer/vectorizer.py

```diff
@@ -319,6 +328,89 @@ def insert_errors_query(self) -> sql.Composed:
             self.errors_table_ident,
         )
 
+    @cached_property
+    def fetch_batches_to_process_query(self) -> sql.Composed:
+        return sql.SQL("""
+            SELECT openai_batch_id, output_file_id FROM {}.{}
+            WHERE status not in('failed', 'processed', 'prepared')
```

Comment:
This makes sense.

Does that also mean changing the implementation so that only in-progress batches are stored in the database, getting rid of storing and evaluating the status? I noticed your query does not contain the `WHERE status not in('failed', 'processed', 'prepared')` anymore.

Reply:
I've now implemented the changes to the query as per your suggestion.